src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -38,27 +38,33 @@ public class LectureUnitProcessingService {
 
     private final AttachmentUnitService attachmentUnitService;
 
+    private final PDFTextStripper pdfTextStripper;
+
+    // A pdf splitter that should be used to split a file into single pages
+    private final Splitter pdfSinglePageSplitter;
+
     public LectureUnitProcessingService(SlideSplitterService slideSplitterService, FileService fileService, LectureRepository lectureRepository,
             AttachmentUnitService attachmentUnitService) {
         this.fileService = fileService;
         this.slideSplitterService = slideSplitterService;
         this.lectureRepository = lectureRepository;
         this.attachmentUnitService = attachmentUnitService;
+        this.pdfTextStripper = new PDFTextStripper();
+        this.pdfSinglePageSplitter = new Splitter();
```

Comment:
Is there a reason why you put the initialisation into the constructor instead of just initialising the field directly?